packages/server/src/arpa_reporter/lib/audit-report.js

```diff
@@ -154,24 +158,73 @@ async function generate(requestHost) {
     XLSX.utils.book_append_sheet(workbook, sheet2, 'Project Summaries');
 
     return {
+        periodId,
         filename: `audit report ${moment().format('yy-MM-DD')}.xlsx`,
         outputWorkBook: XLSX.write(workbook, { bookType: 'xlsx', type: 'buffer' }),
     };
 }
 
+function getS3Client() {
+    /*
+        TODO: Move this client generation to a separate module that handles creating all AWS clients.
+        Captured in ticket here: https://github.com/usdigitalresponse/usdr-gost/issues/1161
+    */
+
+    let s3;
+    if (process.env.LOCALSTACK_HOSTNAME) {
+        /*
+            1. Make sure the local environment has awslocal installed.
+            2. Use the commands to create a bucket to test with.
+                - awslocal s3api create-bucket --bucket arpa-audit-reports --region us-west-2 --create-bucket-configuration '{"LocationConstraint": "us-west-2"}'
+            3. Access bucket resource metadata through the following URL.
+                - awslocal s3api list-buckets
+                - awslocal s3api list-objects --bucket arpa-audit-reports
+        */
+        console.log('------------ USING LOCALSTACK ------------');
+        const endpoint = new AWS.Endpoint(`http://localstack:${process.env.EDGE_PORT}`);
+        s3 = new AWS.S3({
+            region: process.env.AWS_DEFAULT_REGION,
+            endpoint,
+            s3ForcePathStyle: true,
+        });
+    } else {
+        s3 = new AWS.S3();
+    }
+    return s3;
+}
+
+async function presignAndSendEmail(Key, recipientEmail) {
+    const s3 = module.exports.getS3Client();
+    // Generate presigned url to get the object
+    const signingParams = { Bucket: AUDIT_REPORT_BUCKET, Key, Expires: SEVEN_DAYS_IN_SECONDS };
+    const signedUrl = s3.getSignedUrl('getObject', signingParams);
+    // Send email once signed URL is created
+    email.sendAuditReportEmail(recipientEmail, signedUrl);
```

Comment:
Were you still planning on updating this to send an API URL with a new redirect endpoint, or holding off on that for now?

Reply:
I'm working on it in a separate branch. It needed a bit more refactoring so its a taking a bit to change up the tests. I'll have that PR ready this afternoon, but wanted to merge this so we can see if there's additional terraform changes needed in case the general flow is failing in staging.